src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -2124,8 +2124,8 @@ public void setCanBeAutoDeleted(final Boolean canBeDeleted)
 
     public BlockPos getRandomOutsiderInDirection(final EnumFacing directionX, final EnumFacing directionZ)
     {
-        final List<BlockPos> positions = wayPoints.keySet().stream().filter(pos -> isInDirection(directionX, directionZ, center.subtract(pos))).collect(Collectors.toList());
-        positions.addAll(buildings.keySet().stream().filter(pos -> isInDirection(directionX, directionZ, center.subtract(pos))).collect(Collectors.toList()));
+        final List<BlockPos> positions = wayPoints.keySet().stream().filter(pos -> isInDirection(directionX, directionZ, pos.subtract(center))).collect(Collectors.toList());
+        positions.addAll(buildings.keySet().stream().filter(pos -> isInDirection(directionX, directionZ, pos.subtract(center))).collect(Collectors.toList()));
```

Comment:
if you flip it anyways, why not use a Math.abs on the result? That way you get the distance

Reply:
because I want the vector in which direction it goes and not the distance =D